pennylane/transforms/decompose.py

```diff
@@ -403,10 +450,11 @@ def decompose(
 
     Args:
         tape (QuantumScript or QNode or Callable): a quantum circuit.
-        gate_set (Iterable[str or type] or Callable, optional): The target gate set specified as
-            either (1) a sequence of operator types and/or names or (2) a function that returns
-            ``True`` if the operator belongs to the target gate set. Defaults to ``None``, in which
-            case the gate set is considered to be all available :doc:`quantum operators </introduction/operations>`.
+        gate_set (Iterable[str or type], Dict[type or str, float], or Callable, optional): The target gate
+            set specified as either (1) a sequence of operator types and/or names, (2) a function that returns
+            ``True`` if the operator belongs to the target gate set or (3) a dictionary of operator
+            types and/or names with weights. Defaults to ``None``, in which case the gate set is
+            considered to be all available :doc:`quantum operators </introduction/operations>`.
         max_expansion (int, optional): The maximum depth of the decomposition. Defaults to None.
```

Comment:
```suggestion
        max_expansion (int, optional): The maximum depth of the decomposition. Defaults to ``None``.
```
Not related to this PR, but something I realized while reviewing the documentation : )